src/disposables.ts

```diff
@@ -0,0 +1,3 @@
+export interface IDisposable {
+  dispose: () => void;
```

Comment:
Note to self: why do we need a `Disposable` interface?

The Disposable is typical from .NET world where you need to release native resources that cannot be garbage collected without an explicit signal, such as `ADO.NET connections` or `IDbContext` from EntityFramework. Javascript does not have this problem at all.

Reply:
The disposable pattern is very useful outside of garbage collection.

```js
const foo = client.startPolling();
// ... later
foo.dispose();
```

While you can, of course, stop the polling in any number of ways, consolidating around a single disposer pattern makes scaling systems with large numbers of disposable "things/processes" easier.

Reply:
Ok ok I got it — `Dispose` as a general lifecycle method rather than disposing native resources. I'm ok with that, we just need to make sure everybody in the company knows we have such general interface.